frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -692,7 +693,30 @@ void main(List<String> args) async {
     });
   });
 
-  group('opaque type', () {
+  group('dart opaque type', () {
+    f() => 'Test_String';
+
+    test('loopback', () async {
+      var back1 = await api.loopBack(notTemp: f) as String Function();
+      expect(back1(), 'Test_String');
+      var back2 = await api.loopBack(notTemp: back1) as String Function();
+      expect(back2(), 'Test_String');
+      expect(identical(back2, f), isTrue);
+    });
+
+    test('drop', () async {
+      expect(await api.asyncDartOpaque(notTemp: f), 'async test');
+      expect(api.syncDartOpaque(notTemp: f), 'test');
+    });
+
+    test('unwrap', () async {
+      var data = 'Test';
+      expect(api.unwrapDartOpaque(opaque: data), 'Test');
+      await expectLater(() => api.panicUnwrapDartOpaque(opaque: data), throwsA(isA<FfiException>()));
```

Comment:
would be great to check `data` is GCed later. (using the boilerplate I will give)

Reply:
(same as above)